packages/dashboard/src/testing/testing-ground/testing-ground.tsx

```diff
@@ -20,6 +111,79 @@ export class TestingGround {
             <iot-dashboard dashboardConfiguration={this.dashboardConfiguration}></iot-dashboard>
           </div>
           <div slot="right">
+            <div>
+              <label htmlFor="selectWidget"> Select widget:</label>
+              <select id="selectWidget" onChange={this.selectWidget}>
+                {this.dashboardConfiguration.widgets.map((widget) => (
+                  <option value={widget.id}>{widget.id.toString() + ', ' + widget.componentTag.toString()}</option>
+                ))}
+              </select>
+            </div>
+            <div>
+              <label>Only list visible annotations</label>
+              <input
+                type="checkbox"
+                checked={this.onlyListVisibleAnnotations}
+                onChange={this.onListVisibleAnnotations}
+              />
+            </div>
+            <br />
+            <br />
+            <div>
+              <form onSubmit={this.addAnnotation}>
+                <label htmlFor="addAnnotation"> Add annotation -</label>
+                <label htmlFor="addLabel"> Label:</label>
+                <input type="text" id="addLabel"></input>
+                <input type="color" value="#FF0000"></input>
+                <input type="datetime-local" step="1"></input>
+                <input type="submit" value="Add" id="addAnnotation" onSubmit={this.addAnnotation}></input>
+              </form>
+            </div>
+            <br />
+            <br />
+            <div>
+              <form onSubmit={this.deleteAnnotation}>
+                <label htmlFor="deleteAnnotation"> Delete annotation:</label>
+                <select id="deleteAnnotation">
+                  {this.onlyListVisibleAnnotations
```

Comment:
This is a lot of logic to be stuffing into a single JSX expression - Can we move this up at least to above the render function's return, and make it a little clearer by refactoring to successive const declarations rather than one big nested ternary?

Reply:
Refactored; moved filter and find logic above the render function